weather_dl/download_pipeline/parsers.py

```diff
@@ -326,6 +333,29 @@ def _number_of_replacements(s: t.Text):
     return len(set(format_names)) + num_empty_names
 
 
+def _validate_target_path(config: Config):
+    """Checks accessibility of target locations before execution start"""
+    for partition_conf in prepare_partitions(config):
+        target = prepare_target_name(partition_conf)
+        parsed = urlparse(target)
+        if FileSystems.exists(target):
+            raise ValueError(f"Path {target} already exists.")
+        try:
+            if parsed.scheme == "gs":
+                GcsIO().open(target, 'w').close()
+            elif parsed.scheme == "s3":
+                S3IO().open(target, 'w').close()
+            elif parsed.scheme == "" or parsed.scheme == "fs":
+                open(target, 'w').close()
+            else:
+                logger.warning(f"Unable to verify path {target} writeable."
+                               f"Persistent IO errors when writing to the target"
+                               f"path during execution may cause deadlocks.")
+        except Exception:
+            raise ValueError(f"Unable to write to {target}")
```

Comment:
One more idea: let's chain the errors:
```suggestion
        except Exception as e:
            raise ValueError(f"Unable to write to {target}") from e
```